src/engine/douglas_peucker.cpp

```diff
@@ -59,13 +59,15 @@ struct CoordinatePairCalculator
 };
 }
 
-void DouglasPeucker::Run(std::vector<SegmentInformation> &input_geometry, const unsigned zoom_level)
+void DouglasPeucker(std::vector<SegmentInformation>::iterator begin,
```

Comment:
Urgh yeah have to get used to that again. Also having different conventions between member functions and free standing function is somewhat ugly for refactoring :/